docs/dev/object-deletion.md

```diff
@@ -0,0 +1,118 @@
+
+If an object cannot be successfully removed CVO will set `Upgradeable=False` which in turn blocks cluster update to the next minor release.
+
+## Examples
+
+The following examples provide guidance to OpenShift developers on how resources may be removed but this will vary depending on the component.
+Ultimately it is the developer's responsibility to ensure the removal works by thoroughly testing.
+In all cases, and as general guidance, an operator should never allow itself to be removed if the operator's operand has not been removed.
+
+### The autoscaler operator
+
+Remove the cluster-autoscaler-operator deployment.
+The existing cluster-autoscaler-operator deployment manifest 0000_50_cluster-autoscaler-operator_07_deployment.yaml is modified to contain the delete annotation:
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler-operator
+  namespace: openshift-machine-api
+  annotations:
+    release.openshift.io/delete: "true"
+...
+```
+Additional manifest properties such as `spec` may be set if convenient (e.g. because you are looking to make a minimal change vs. a previous version of the manifest), but those properties have no affect on manifests with the delete annotation.
+
+### The service-catalog operators
+
+In release 4.5 two jobs, openshift-service-catalog-controller-manager-remover and openshift-service-catalog-apiserver-remover, were created to remove the Service Catalog.
+In release 4.6, these jobs and all their supporting cluster objects also needed to be removed.
+This example shows how Service Catalog removal would have been executed had manifest based object deletion been in place.
+
+The Service Catalog is composed of two components, the cluster-svcat-apiserver-operator and the cluster-svcat-controller-manager-operator.
+Each of these components use manifests for creation/update of the component's required resources: namespace, roles, operator deployment, etc.
+The cluster-svcat-apiserver-operator had [the following associated manifests][svcat-apiserver-4.4-manifests]:
+
+* `0000_50_cluster-svcat-apiserver-operator_00_namespace.yaml` containing the `openshift-service-catalog-apiserver-operator` namespace.
+    The deletion annotation would be added to this manifest.
+* `0000_50_cluster-svcat-apiserver-operator_02_config.crd.yaml` containing a cluster-scoped CRD.
+    The deletion annotation would be added to this manifest.
+* `0000_50_cluster-svcat-apiserver-operator_03_config.cr.yaml` containing a cluster-scoped, create-only ServiceCatalogAPIServer.
+    The deletion annotation would be added to this manifest.
+* `0000_50_cluster-svcat-apiserver-operator_03_configmap.yaml` containing a ConfigMap in the `openshift-service-catalog-apiserver-operator` namespace.
```

Comment:
We should say ` The deletion annotation needs be added to these manifests below` and then list all the manifests that need the annotation.

Reply:
done